src/pokemonClass.ts

```diff
@@ -0,0 +1,69 @@
+export type pokemonType =
+  | "grass"
+  | "poison"
+  | "fire"
+  | "water"
+  | "bug"
+  | "normal"
+  | "flying"
+  | "electric"
+  | "ground"
+  | "fairy"
+  | "psychic"
+  | "fighting"
+  | "rock";
+
+export abstract class Pokemon {
+  constructor() {}
+  abstract readonly _name: string;
+  abstract readonly _image: string;
+  abstract readonly _maxHP: number;
+  abstract _currentHP: number;
+  abstract _attackPoints: number;
+  abstract _defensePoints: number;
+  abstract readonly _types: pokemonType[];
+  abstract readonly _moves: { moveName: string; moveType: pokemonType }[];
+  abstract _isAlive: boolean = true;
```

Comment:
there is no need for another property for that :)
What would happen if currentHP would be equal 0, but isAlive would be true? which one would be a source of truth?